### ISSUE TITLE
Can't change user's container

### ISSUE THREAD

**Issue by chenboshiun0109** [hashicorp/terraform-provider-ad#48]
### Terraform Version and Provider Version

* Terraform v0.13.3

### Windows Version

Windows Server 2016 Datacenter

### Affected Resource(s)

```
resource "ad_user" "u" {
  principal_name   = var.principal_name
  sam_account_name = var.samaccountname
  display_name     = "Terraform Test User"
  container        = "OU=Test,OU=Terraform,DC=tester,DC=id"
  cannot_change_password = false
  password_never_expires = false
  enabled          = true
```

### Terraform Configuration Files

```
variable principal_name { default = "testuser@tester.com" }
variable samaccountname { default = "testuser" }

resource "ad_ou" "o_terraform" {
    name = "Terraform"
    path = "dc=tester,dc=id"
    description = "OU for gplink tests"
    protected = false
}

resource "ad_ou" "o_test" {
    name = "Test"
    path = "ou=Terraform,dc=tester,dc=id"
    description = "OU for gplink tests"
    protected = false
}

resource "ad_user" "u" {
  principal_name   = var.principal_name
  sam_account_name = var.samaccountname
  display_name     = "Terraform Test User"
  container        = "OU=Test,OU=Terraform,DC=tester,DC=id"
  cannot_change_password = false
  password_never_expires = false
  enabled          = true
}
```

### Expected Behavior

Terraform will perform the following actions:

```
  # ad_user.u will be updated in-place
  ~ resource "ad_user" "u" {
        cannot_change_password = false
      ~ container              = "OU=Testing Account,DC=tester,DC=id" -> "OU=Test,OU=Terraform,DC=tester,DC=id"
        display_name           = "Terraform Test User"
        enabled                = true
        id                     = "6a5b08c6-bdb3-42d5-9938-60affd2d3f5a"
        password_never_expires = false
        principal_name         = "testuser@tester.com"
        sam_account_name       = "testuser"
    }
```

### Actual Behavior

```
Error: command Set-ADUser exited with a non-zero exit code 1, stderr: #< CLIXML
<Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>0</PC><T>Processing</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="1"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>25</PC><T>Processing</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="2"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>50</PC><T>Processing</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="3"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>75</PC><T>Processing</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="4"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>100</PC><T>Processing</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="5"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>100</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="6"><TNRef RefId="0" /><MS><I64 N="SourceId">2</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><S S="Error">Set-ADUser : A parameter cannot be found that matches parameter name 'Path'._x000D__x000A_</S><S S="Error">At line:1 char:61_x000D__x000A_</S><S S="Error">+ ... ADUser -Identity "6a5b08c6-bdb3-42d5-9938-60affd2d3f5a" -Path "OU=Tes ..._x000D__x000A_</S><S S="Error">+                                                             ~~~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : InvalidArgument: (:) [Set-ADUser], ParameterBindingException_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.ActiveDirectory.Management.Commands.SetADUser_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
```
can't mv user's container, show error message: "Set-ADUser : A parameter cannot be found that matches parameter name 'Path'."

### Steps to Reproduce

1. terraform plan

2. Terraform will perform the following actions:

```
  # ad_user.u will be updated in-place
  ~ resource "ad_user" "u" {
        cannot_change_password = false
      ~ container              = "OU=Testing Account,DC=tester,DC=id" -> "OU=Test,OU=Terraform,DC=tester,DC=id"
        display_name           = "Terraform Test User"
        enabled                = true
        id                     = "6a5b08c6-bdb3-42d5-9938-60affd2d3f5a"
        password_never_expires = false
        principal_name         = "testuser@tester.com"
        sam_account_name       = "testuser"
    }
```

3. terraform apply

### Important Factoids
No

### References
No

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by chenboshiun** [comment]
Hello~

Can anyone help me ?

**Comment by koikonom** [comment]
Hi @chenboshiun, 

Can you please reproduce the issue with the environment variable TF_LOG set to `DEBUG` before you run the apply step and send us the logs?

**Comment by chenboshiun** [comment]
Hi @koikonom 

I have exported DEBUG log:

```
2020-10-21T17:00:01.288+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: 2020/10/21 17:00:01 [DEBUG] sanitising key "sam_account_name" to: testuser
2020-10-21T17:00:01.288+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: 2020/10/21 17:00:01 [DEBUG] sanitising key "principal_name" to: testuser@kkbox.com
2020-10-21T17:00:01.288+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: 2020/10/21 17:00:01 [DEBUG] sanitising key "display_name" to: Terraform Test User
2020-10-21T17:00:01.288+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: 2020/10/21 17:00:01 [DEBUG] sanitising key "container" to: OU=Test,OU=Terraform,DC=kkboxer,DC=id
2020-10-21T17:00:01.288+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: 2020/10/21 17:00:01 [DEBUG] sanitising key "initial_password" to:
2020-10-21T17:00:01.288+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: 2020/10/21 17:00:01 Modifying user: "testuser@kkbox.com"
2020-10-21T17:00:01.288+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: 2020/10/21 17:00:01 [DEBUG] Running command Set-ADUser -Identity "6a5b08c6-bdb3-42d5-9938-60affd2d3f5a" -Path "OU=Test,OU=Terraform,DC=kkboxer,DC=id" via powershell
2020-10-21T17:00:01.288+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: 2020/10/21 17:00:01 [DEBUG] Encoded command: powershell.exe -EncodedCommand UwBlAHQALQBBAEQAVQBzAGUAcgAgAC0ASQBkAGUAbgB0AGkAdAB5ACAAIgA2AGEANQBiADAAOABjADYALQBiAGQAYgAzAC0ANAAyAGQANQAtADkAOQAzADgALQA2ADAAYQBmAGYAZAAyAGQAMwBmADUAYQAiACAALQBQAGEAdABoACAAIgBPAFUAPQBUAGUAcwB0ACwATwBVAD0AVABlAHIAcgBhAGYAbwByAG0ALABEAEMAPQBrAGsAYgBvAHgAZQByACwARABDAD0AaQBkACIA
2020-10-21T17:00:01.346+0800 [DEBUG] plugin.terraform-provider-azurerm_v2.20.0_x5: Testing if Service Principal / Client Certificate is applicable for Authentication..
2020-10-21T17:00:01.346+0800 [DEBUG] plugin.terraform-provider-azurerm_v2.20.0_x5: Testing if Multi Tenant Service Principal / Client Secret is applicable for Authentication..
2020-10-21T17:00:01.346+0800 [DEBUG] plugin.terraform-provider-azurerm_v2.20.0_x5: Testing if Service Principal / Client Secret is applicable for Authentication..
2020-10-21T17:00:01.346+0800 [DEBUG] plugin.terraform-provider-azurerm_v2.20.0_x5: Testing if Managed Service Identity is applicable for Authentication..
2020-10-21T17:00:01.346+0800 [DEBUG] plugin.terraform-provider-azurerm_v2.20.0_x5: Testing if Obtaining a token from the Azure CLI is applicable for Authentication..
2020-10-21T17:00:01.346+0800 [DEBUG] plugin.terraform-provider-azurerm_v2.20.0_x5: Using Obtaining a token from the Azure CLI for Authentication
2020-10-21T17:00:01.347+0800 [DEBUG] plugin.terraform-provider-azurerm_v2.20.0_x5: [DEBUG] Resource "https://management.core.windows.net/" isn't for the correct Tenant
2020-10-21T17:00:02.049+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: 2020/10/21 17:00:02 [DEBUG] Powershell command exited with code 1
2020-10-21T17:00:02.049+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: 2020/10/21 17:00:02 [DEBUG] Stdout: , Stderr: #< CLIXML
N="SourceId">2</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><S S="Error">Set-ADUser : A parameter cannot be found that matches parameter name 'Path'._x000D__x000A_</S><S S="Error">At line:1 char:61_x000D__x000A_</S><S S="Error">+ ... ADUser -Identity "6a5b08c6-bdb3-42d5-9938-60affd2d3f5a" -Path "OU=Tes ..._x000D__x000A_</S><S S="Error">+                                                             ~~~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : InvalidArgument: (:) [Set-ADUser], ParameterBindingException_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.ActiveDirectory.Management.Commands.SetADUser_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
2020-10-21T17:00:02.049+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: 2020/10/21 17:00:02 [DEBUG] stderr: #< CLIXML
I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>75</PC><T>Processing</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="4"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>100</PC><T>Processing</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="5"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>100</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="6"><TNRef RefId="0" /><MS><I64 N="SourceId">2</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><S S="Error">Set-ADUser : A parameter cannot be found that matches parameter name 'Path'._x000D__x000A_</S><S S="Error">At line:1 char:61_x000D__x000A_</S><S S="Error">+ ... ADUser -Identity "6a5b08c6-bdb3-42d5-9938-60affd2d3f5a" -Path "OU=Tes ..._x000D__x000A_</S><S S="Error">+                                                             ~~~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : InvalidArgument: (:) [Set-ADUser], ParameterBindingException_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.ActiveDirectory.Management.Commands.SetADUser_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
2020-10-21T17:00:02.049+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: stdout:
2020/10/21 17:00:02 [DEBUG] ad_user.u: apply errored, but we're indicating that via the Error pointer rather than returning it: command Set-ADUser exited with a non-zero exit code 1, stderr: #< CLIXML
eId">2</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><S S="Error">Set-ADUser : A parameter cannot be found that matches parameter name 'Path'._x000D__x000A_</S><S S="Error">At line:1 char:61_x000D__x000A_</S><S S="Error">+ ... ADUser -Identity "6a5b08c6-bdb3-42d5-9938-60affd2d3f5a" -Path "OU=Tes ..._x000D__x000A_</S><S S="Error">+                                                             ~~~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : InvalidArgument: (:) [Set-ADUser], ParameterBindingException_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.ActiveDirectory.Management.Commands.SetADUser_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
2020/10/21 17:00:02 [ERROR] eval: *terraform.EvalApplyPost, err: command Set-ADUser exited with a non-zero exit code 1, stderr: #< CLIXML
I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>75</PC><T>Processing</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="4"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>100</PC><T>Processing</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="5"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Loading Active Directory module for Windows PowerShell with default drive 'AD:'</AV><AI>0</AI><Nil /><PI>-1</PI><PC>100</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="progress" RefId="6"><TNRef RefId="0" /><MS><I64 N="SourceId">2</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><S S="Error">Set-ADUser : A parameter cannot be found that matches parameter name 'Path'._x000D__x000A_</S><S S="Error">At line:1 char:61_x000D__x000A_</S><S S="Error">+ ... ADUser -Identity "6a5b08c6-bdb3-42d5-9938-60affd2d3f5a" -Path "OU=Tes ..._x000D__x000A_</S><S S="Error">+                                                             ~~~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : InvalidArgument: (:) [Set-ADUser], ParameterBindingException_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.ActiveDirectory.Management.Commands.SetADUser_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
2020-10-21T17:00:02.049+0800 [DEBUG] plugin.terraform-provider-ad_v0.2.0_x5: stdout:
2020/10/21 17:00:02 [DEBUG] ad_user.u: apply errored, but we're indicating that via the Error pointer rather than returning it: command Set-ADUser exited with a non-zero exit code 1, stderr: #< CLIXML
eId">2</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><S S="Error">Set-ADUser : A parameter cannot be found that matches parameter name 'Path'._x000D__x000A_</S><S S="Error">At line:1 char:61_x000D__x000A_</S><S S="Error">+ ... ADUser -Identity "6a5b08c6-bdb3-42d5-9938-60affd2d3f5a" -Path "OU=Tes ..._x000D__x000A_</S><S S="Error">+                                                             ~~~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : InvalidArgument: (:) [Set-ADUser], ParameterBindingException_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.ActiveDirectory.Management.Commands.SetADUser_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
2020/10/21 17:00:02 [ERROR] eval: *terraform.EvalApplyPost, err: command Set-ADUser exited with a non-zero exit code 1, stderr: #< CLIXML
</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><S S="Error">Set-ADUser : A parameter cannot be found that matches parameter name 'Path'._x000D__x000A_</S><S S="Error">At line:1 char:61_x000D__x000A_</S><S S="Error">+ ... ADUser -Identity "6a5b08c6-bdb3-42d5-9938-60affd2d3f5a" -Path "OU=Tes ..._x000D__x000A_</S><S S="Error">+                                                             ~~~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : InvalidArgument: (:) [Set-ADUser], ParameterBindingException_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.ActiveDirectory.Management.Commands.SetADUser_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
2020/10/21 17:00:02 [ERROR] eval: *terraform.EvalSequence, err: command Set-ADUser exited with a non-zero exit code 1, stderr: #< CLIXML
/AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><S S="Error">Set-ADUser : A parameter cannot be found that matches parameter name 'Path'._x000D__x000A_</S><S S="Error">At line:1 char:61_x000D__x000A_</S><S S="Error">+ ... ADUser -Identity "6a5b08c6-bdb3-42d5-9938-60affd2d3f5a" -Path "OU=Tes ..._x000D__x000A_</S><S S="Error">+                                                             ~~~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : InvalidArgument: (:) [Set-ADUser], ParameterBindingException_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.ActiveDirectory.Management.Commands.SetADUser_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
```

**Comment by koikonom** [comment]
Thanks for that. This is definitely a bug and it will be addressed in the next provider release.

**Comment by koikonom** [comment]
Fixed in v0.3.0.
